include/core/mir/fatal.h

```diff
@@ -31,6 +31,7 @@ namespace mir
  * Note the reason parameter is a simple char* so its value is clearly visible
  * in stack trace output.
  * \remark There is no attempt to make this thread-safe, if it needs to be changed
+ * \remark Since MirCore 0.25
```

Comment:
I don't think we want (nor should we need) "Since" annotations that date from before the current major version/SONAME.

Reply:
What exactly are you suggesting? Are you saying you prefer the current system of maintaining a separate file with symbols, or do you just want a different type of in-code annotation?

This PR comes out of discussions with RAOF where we concluded keeping export info alongside the symbols in the code would be nicer than in a separate file (in which case, all exported functions need to be annotated somehow). I suppose the generator could read a symbols.map file and leave previous stanzas as-is, but I like the idea of the generator being a pure function from source code to symbols.

Reply:
> it seems the script could take a “what is the MirAL SOVER” parameter, and use that as the version for all symbols either without “Since” annotations

Reply:
It's less about the version, and more about knowing which symbols to export at all. If the version seems like clutter we could make a different annotation that doesn't include the version but still marks them for export.

Reply:
> It's less about the version, and more about knowing which symbols to export at all. If the version seems like clutter we could make a different annotation that doesn't include the version but still marks them for export.

It isn't obvious why annotations are required, or even that they are a good, solution to this.

Symbols should be exported if they are:
1. Declared in a public header;
2. Accessible (i.e. namespace scope, public/protected static members, public/protected member functions);
3. Not inline, nor template

Reply:
Oooh, yeah. That's a nice approach!

There's the chance with 3. that they *do* need to be exported, but that's only for weird scenarios (eg: where something takes the address of the function) and I don't *think* we rely on any of that behaviour.

It seems that Doxygen collects that information, how hard would it be to use it?

Reply:
Not very, that's used in `regenerate-miral-symbols-map.py` already